common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -480,12 +491,15 @@ private boolean sendReportWorker(boolean explicitRequest) {
     void addSpan(Span span) {
         lastNewSpanMillis.set(System.currentTimeMillis());
 
+        if (client == null) {
+            initializeCollectorClient();
```

Comment:
This can now call initialize a large # of times if the client is null when several spans are finished.